crates/settings/src/network.rs

```diff
@@ -45,9 +47,81 @@ impl Network {
         }
     }
 
+    pub fn parse_networks_from_yaml(
+        document: Arc<RwLock<StrictYaml>>,
+    ) -> Result<HashMap<String, Network>, YamlError> {
+        let document_read = document.read().map_err(|_| YamlError::ReadLockError)?;
+
+        let networks_hash = require_hash(
+            &document_read,
+            Some("networks"),
+            Some("missing field: networks".to_string()),
+        )?;
+
+        let networks = networks_hash
+            .into_iter()
+            .map(|(key, value)| {
+                let key = key.as_str().unwrap_or_default().to_string();
+                let network = Network {
+                    document: document.clone(),
+                    name: key.clone(),
+                    rpc: Url::from_str(&require_string(
+                        &value,
+                        Some("rpc"),
+                        Some(format!("rpc string missing in network: {key}")),
+                    )?)
+                    .map_err(|e| ParseNetworkConfigSourceError::RpcParseError(e))?,
+                    chain_id: require_string(
+                        &value,
+                        Some("chain-id"),
+                        Some(format!("chain-id string missing in network: {key}")),
```

Comment:
i think saying string here is a bit confusing, because it's also an error in the next step if it's a string (because it can't be parsed to a u64). from the user perspective if they put "one", that's not valid, but this error indicates it would be